src/pages/PuzzlePage/index.tsx

```diff
@@ -0,0 +1,52 @@
+import { MultiLinkLayer } from "@/components/Layout";
+import { MockGameDataTypes } from "@/types/linkToGame";
+import { useSearchParams } from "react-router-dom";
+import { TimeTrekFrame } from "./components/TimeTrek";
+import { TimeTriviaFrame } from "./components/TimeTrivia";
+// import { useState } from "react";
+const gameListData: MockGameDataTypes[] = [{

```

Comment:
more more modular approach, also store the component in the gameListData, and then render the game according to the selected game, 

maybe you can do something like: 
```
...
linkTo: {
     timeTrek: {
          id: 1,
          title: ..
          component: ..
     },
     ...
}
```


render `gameListData.linkTo.gameParams.component`